test/e2e_node/benchmark_test.go

```diff
@@ -0,0 +1,168 @@
+		cl = client.NewOrDie(&restclient.Config{Host: *apiServerAddress})
+	})
+
+	Describe("container benchmark blackbox test", func() {
+		Context("when list testing images", func() {
+			var conformImages []ConformanceImage
+			conformImageTags := []string{
+				"gcr.io/google_containers/busybox",
+				"gcr.io/google_containers/nginx",
+			}
+			Measure("it should pull successfully [Performance]", func(b Benchmarker) {
+				b.Time("pull images", func() {
+					for _, imageTag := range conformImageTags {
+						image, _ := NewConformanceImage("docker", imageTag)
+						conformImages = append(conformImages, image)
+
+						for start := time.Now(); time.Since(start) < imageRetryTimeout; time.Sleep(imagePullInterval) {
+							if err := image.Pull(); err == nil {
+								break
+							}
+						}
+					}
+				})
+			}, 1)
+			Measure("it should list images successfully [Performance]", func(b Benchmarker) {
+				b.Time("list images", func() {
+					RunParallel(func(id int) error {
+						image, _ := NewConformanceImage("docker", "")
+						_, err := image.List()
+						return err
+					}, concurrencyCount)
+				})
+			}, iterCount)
+
+			Measure("it should remove successfully [Performance]", func(b Benchmarker) {
+				b.Time("remove images", func() {
+					for _, image := range conformImages {
+						image.Remove()
+					}
+				})
+			}, 1)
+		})
+
+		Context("when running containers concurrency", func() {
+			Measure("it should start containers successfully [Performance]", func(b Benchmarker) {
+				b.Time("start containers", func() {
+					RunParallel(func(id int) error {
+						containerName := fmt.Sprintf("busybox-%d", id)
+						container := ConformanceContainer{
+							Container: api.Container{
+								Image:           "gcr.io/google_containers/busybox",
+								Name:            containerName,
+								Command:         []string{"sh", "-c", "env"},
+								ImagePullPolicy: api.PullIfNotPresent,
+							},
+							Client:   cl,
+							NodeName: *nodeName,
+						}
+						if err := container.Create(); err != nil {
+							return err
+						}
+
+						for start := time.Now(); time.Since(start) < retryTimeout; time.Sleep(pollInterval) {
+							if pod, err := container.Get(); err != nil {
+								return err
+							} else if pod.Phase != api.PodPending {
+								return nil
+							}
+						}
+
+						return errors.New("Pending to start container")
+					}, concurrencyCount)
+				})
+				b.Time("get container", func() {
+					RunParallel(func(id int) error {
+						containerName := fmt.Sprintf("busybox-%d", id)
+						container := ConformanceContainer{
+							Container: api.Container{
+								Name: containerName,
+							},
+							Client: cl}
+						_, err := container.Get()
+						return err
+					}, concurrencyCount)
+				})
+				b.Time("list containers", func() {
+					RunParallel(func(id int) error {
+						container := ConformanceContainer{Client: cl}
+						_, err := container.List()
+						return err
+					}, concurrencyCount)
```

Comment:
I think the `Get()` and `List()` here are mainly testing the apiserver instead of kubelet.